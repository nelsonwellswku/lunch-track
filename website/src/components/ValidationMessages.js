import React from 'react';

const ValidationMessages = ({ errors } = {}) => {
  if (!errors.length) { return null; }

  return (
    <div className="alert alert-danger">
      <ul>
        {errors.map(x => <li key={x}>{x}</li>)}
      </ul>
    </div>
  );
};

export default ValidationMessages;
