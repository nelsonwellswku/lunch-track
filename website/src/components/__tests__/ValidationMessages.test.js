import React from 'react';
import ValidationMessages from '../ValidationMessages';
import { shallow } from 'enzyme';

describe('validation messages component', () => {
  test('returns nothing if no props are passed', () => {
    const wrapper = shallow(<ValidationMessages />);
    expect(wrapper).toMatchSnapshot();
  });

  test('returns an unordered list of error messages', () => {
    const errors = [
      'This is error one',
      'This is error two',
    ];

    const wrapper = shallow(<ValidationMessages errors={errors} />);
    expect(wrapper).toMatchSnapshot();
  });
});
