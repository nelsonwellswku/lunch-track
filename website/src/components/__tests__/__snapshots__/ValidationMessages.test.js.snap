// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validation messages component returns an unordered list of error messages 1`] = `
<div
  className="alert alert-danger"
>
  <ul>
    <li
      key="This is error one"
    >
      This is error one
    </li>
    <li
      key="This is error two"
    >
      This is error two
    </li>
  </ul>
</div>
`;

exports[`validation messages component returns nothing if no props are passed 1`] = `""`;
